scripts/nested-libvirt/router-check.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+set -e
+set -u
+set -o pipefail
+
+# TODO: add ability to pass cluster-name
+# wait for the router namespace
+KUBECONFIG="${HOME}"/clusters/nested/auth/kubeconfig
+API_UP=
+ROUTER_NAMESPACE=
+ROUTER_DEPLOYMENT=
+while true; do
+  if [[ ! -f "${KUBECONFIG}" ]]; then
+    sleep 15 & wait
+    continue
+  fi
+  if ! oc get nodes 2>/dev/null; then
+    echo "Waiting for API at $(oc whoami --show-server) to respond ..."
+    sleep 15 & wait
+    continue
+  elif [[ -z "${API_UP}" ]]; then
+    echo "API at $(oc whoami --show-server) has responded"
+    API_UP=1
+  fi
+  if [[ -z "${ROUTER_NAMESPACE}" ]]; then
+    # check multiple namespaces while we are transitioning to the new locations
+    if oc get deploy/router -n tectonic-ingress 2>/dev/null; then
+      ROUTER_NAMESPACE=tectonic-ingress
+      ROUTER_DEPLOYMENT="deploy/router"
+      elif oc get ds/router-default -n openshift-ingress 2>/dev/null; then
+        ROUTER_NAMESPACE=openshift-ingress
+        ROUTER_DEPLOYMENT="ds/router-default"
+      elif oc get deploy/router -n openshift-ingress 2>/dev/null; then
+        ROUTER_NAMESPACE=openshift-ingress
+        ROUTER_DEPLOYMENT="deploy/router"
+      elif oc get deploy/router -n default 2>/dev/null; then
+        ROUTER_NAMESPACE=default
+        ROUTER_DEPLOYMENT="deploy/router"
+      else
+        echo "Waiting for router to be created ..."
+        sleep 15 & wait
+        continue
+      fi
+    echo "Found router in ${ROUTER_NAMESPACE}"
+  fi
+  break
```

Comment:
You can shift this `break` up to put it after `echo "Found router ..."` (or just add another `break` there).

And it looks like there's no timeout cap on this loop?

Reply:
I've added a 10 min timeout.  This script runs after the install completes successfully.  We can tweak the test to determine cluster health, but for now the script closely resembles e2e-aws, that checks that worker is up, router is deployed.